luigi/execution_summary.py

```diff
@@ -0,0 +1,370 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2015-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module provide the function :py:func:`summary` that is used for printing
+an execution summary at the end of luigi invocations. For example:
+
+    .. code:: console
+
+            $ luigi --module examples.execution_summary_example examples.EntryPoint --local-scheduler
+            ...
+            ... lots of spammy output
+            ...
+            INFO: There are 11 pending tasks unique to this worker
+            INFO: Worker Worker(salt=843361665, workers=1, host=arash-spotify-T440s, username=arash, pid=18534) was stopped. Shutting down Keep-Alive thread
+            INFO:
+           ===== Luigi Execution Summary =====
+
+            Scheduled 210 tasks of which:
```

Comment:
Isn't it a bit confusing to users to encounter the notion of "scheduling of tasks that are already done"? The summary could be more business logic friendly. E.g. skip saying "scheduled" at all, and say "195 present dependencies were encountered", or something like that...